src/Azure/Azure.Quantum.Client/SubmitterFactory.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+namespace Microsoft.Azure.Quantum
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using System.Reflection;
+    using System.Text.RegularExpressions;
+    using Microsoft.Quantum.Runtime.Submitters;
+
+    /// <summary>
+    /// Creates Azure Quantum submitters for specific targets.
+    /// </summary>
+    public static class SubmitterFactory
+    {
+        /// <summary>
+        /// Information about each supported QIR submitter.
+        /// </summary>
+        private static readonly ImmutableList<SubmitterInfo> QirSubmitters = ImmutableList.Create(
+            new SubmitterInfo(
+                new Regex(@"^microsoft\.simulator\.([\w]+\.)*[\w]+$"),
```

Comment:
Very minor. Consider using \A and \z (instead of ^ and $) as they always match the beginning and the end of the string.

Reply:
Changed to \A and \z.